Core/src/com/serotonin/m2m2/db/H2Proxy.java

```diff
@@ -422,4 +360,115 @@ public boolean tableExists(ExtendedJdbcTemplate ejt, String tableName) {
                 new Object[] {}, Integer.class, 0) > 0;
     }
     
+    /**
+     * Get the current version of the database.
+     * @return
+     */
+    public int getCurrentVersion() {
+        return Constants.BUILD_ID;
+    }
+
+    /**
+     * Get the expected runtime database name
+     * @return
+     */
+    public String getDatabaseFileSuffix() {
+        return Constants.BUILD_ID + ".h2.db";
+    }
+    
+    /**
+     * Parse the version out of a database name, the format should be
+     * *.[version].h2.db
+     *   This method takes into account that there may be extra periods in the name
+     * @param databaseName
+     * @return
+     * @throws NumberFormatException
+     */
+    public int getVersion(String databaseName) throws IOException {
+        String[] parts = databaseName.split("\\.");
+        try{
+            return getVersion(parts);
+        }catch(NumberFormatException e) {
+            throw new IOException("Invalid h2 database name format: " + databaseName);
+        }
+    }
+    
+    private static int getVersion(String[] parts) throws NumberFormatException {
+        return Integer.parseInt(parts[parts.length - 3]);
+    }
+    
+    /**
+     * Check the name of the database file to see if it is a legacy db.
+     * 
+     * *.h2.db = legacy
+     * *.[version].h2.db = current
+     * 
+     * @return
+     */
+    public static boolean isLegacy(String databaseName) throws IOException {
```

Comment:
This function seems wrong. The else clause checks if it can get a version from a function happy to throw a NumberFormatException) and then still returns true, though.

Reply:
Agreed fixed.